lib/internal/streams/BufferList.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Buffer = require('buffer').Buffer;
+
+module.exports = BufferList;
+
+function BufferList() {
+  this.head = null;
+  this.tail = null;
+  this.length = 0;
+}
+
+BufferList.prototype.push = function(v) {
+  const entry = { data: v, next: null };
```

Comment:
Do you think it could help to create an `Entry` class to increase property access performance?